### PR TITLE
Pull request for new rule wizard (issue #17)

### PR DESCRIPTION
**Pull request by gokcekrky**

### CODE REVIEW

File: de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/utils/ClassLoader.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+public class ClassLoader {
+
+	public static Class<?> LoadFromJar(String keyClass,String jarPath) {
+	     URL jarUrl;
+		try {
+			jarUrl = new File(jarPath).toURI().toURL(); 
+			URLClassLoader loader = new URLClassLoader(new URL[]{jarUrl});
+			JarFile jar;
+			try {
```
Comment (kruegers) [inline de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/utils/ClassLoader.java]: Please don't nest try and catch blocks.

File: de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/utils/ClassLoader.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+				boolean found = false;
+				while (e.hasMoreElements() && !found) {
+		            JarEntry je = (JarEntry) e.nextElement();
+		            if(je.isDirectory() || !je.getName().endsWith(".class")){
+		                continue;
+		            }
+		            String classname =
+		                    je.getName().replace('/', '.').substring(0, je.getName().length() - 6).split("\\$")[0];
+		            if (classname.equals(keyClass)){
+		            	 try {
```
Comment (kruegers) [inline de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/utils/ClassLoader.java]: See comment above.

File: de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/utils/ClassLoader.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+							return c;
+						} catch (ClassNotFoundException e1) {
+							// TODO Auto-generated catch block
+							System.out.println(classname + "is not found");
+						}
+		            	 found = true;
+		            }
+				}
+			} catch (IOException e1) {
+				// TODO Auto-generated catch block
```
Comment (kruegers) [inline de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/utils/ClassLoader.java]: Remove todos if you no longer intend on addressing them.

File: de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/utils/ClassPathSolver.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		}
+
+		return libraryClassPath;
+	}
+
+	private static void resolveClassPathEntry(IClasspathEntry entry, Collection<String> libraryClassPath, IJavaProject project) {
+		IClasspathEntry[] rentries;
+		switch (entry.getEntryKind()) {
+		case IClasspathEntry.CPE_SOURCE:
+			//libraryClassPath.addAll(applicationClassPath(project));
```
Comment (kruegers) [inline de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/utils/ClassPathSolver.java]: Is that line commented intentionally?

File: de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/wizard/CryptSLFile.java [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+							abbr = abbr+ Character.toString(randomalph.charAt(r.nextInt(randomalph.length())));
+							abbr = Character.toLowerCase(abbr.charAt(0))+abbr.substring(1);
+							abbrList.add(abbr);
+						}
+					}
+				}
+				
+			}
+			//
+			/*abbr = Character.toString(methodName.charAt(0));
```
Comment (kruegers) [inline de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/wizard/CryptSLFile.java]: If the block is no longer needed, please remove it.

**Comment by gokcekrky** [comment]
Hi @kruegers , 
I made requested changes. Can you please have a look.

File: de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/utils/ClassPathSolver.java (hunk 4 above)
Comment (gokcekrky) [reply de.darmstadt.tu.crossing.CryptSL.ui/src/de/darmstadt/tu/crossing/ui/utils/ClassPathSolver.java]: I mirror this class from cognicrypt- sootrunner but in my implementation it doesn't reach this line.
So I commented out it first. After your change request I added it again.

**Comment by kruegers** [comment]
Thanks for fixing that. I've taken a look at the feature itself and would request a few more changes before merging.
grafik
1. CryptSL -> CrySL (see #20).
2. Please add the CogniCrypt logo to the  entry of "CrySL File".

grafik
3. Switch the entries for class name and name.
4. Rename name to file name.
5. Split folder into two lines: 1.Project and 2. Folder -> Have a browse button for both and with the one for the latter only showing folders in the selected project. The button should also be grayed out when no project is selected.
6. Add a content proposal to class name that is based on the selected project's class path.
7. The default name of (file) name should be the last segment of class name (classname == A.B....N => filename = N) => Once the user changes the default file name, the auto-updating gets disabled and only when the remove the entire file name, it gets enabled again.
7. Modify the first line to say CrySL.
8. By means of tooltips, add short explanations to each of the fields that describe the purpose of that field. For the file name, briefly explan the mechanism described in 7. 

The following screenshot is from the auto-generated rule for java.security.MessageDigest.
grafik
8. The first parameter of isEqual, digest, and update are mapped to the same object byte1. They should instead map to different objects. I would rather have the user change this from several to one objects than the other way around.
9. Please uncomment ORDER and *. To this end, you need to update the grammar to allow * to be a valid expression in the order section. 

The rest of the feature works great. Thanks for working on it.

**Comment by gokcekrky** [comment]
Hi @kruegers,
I've created a new pull request (#29 ) for this issue. 
Can you please consider the new pull request.

**Pull request closed without merging**